templates/client_properties.erb

```diff
@@ -8,13 +8,13 @@
 <%- if scope[pfx + 'simianarmy_client_aws_assume_role_arn'] != nil -%>
   <%--%>simianarmy.client.aws.assumeRoleArn = <%= scope[pfx + 'simianarmy_client_aws_assume_role_arn'] %>
 <%- end -%>
-<%- if scope[pfx + 'simianarmy_client_aws_proxy_host'] != nil -%>
+<%- if scope[pfx + 'simianarmy_client_aws_proxy_host'] != nil and scope[pfx + 'simianarmy_client_aws_proxy_host'] != '' -%>
```

Comment:
Assuming the check is done in ruby, can you try `!foobar.nil? && !foobar.empty?` instead?